misc/prototypes.h

```diff
@@ -134,6 +134,8 @@ char *strchr(const char *s, char c);
 char *strrchr(const char *s, char c);
 char *strchrnul(const char *s, char c);
 
+char *memmem(const char *haystack, size_t haystacklen, const char *needle, size_t needlelen);
```

Comment:
Why do you make it `char *` instead of `void *`?

Reply:
As far as I know, `memmem` is used to handle null-contained strings unlike `strstr` so I made it `char*`. In the man page description, it also says as follows:

> The memmem() function finds the start of the first occurrence of the "substring" needle of length needlelen in the memory area haystack of length haystacklen.

I see the word "substring" here.

Reply:
No it's a general function that can be used with binary data.  With above, it'll always treat it as a string.  Also original prototype also uses `void *`.

Reply:
I was going to show some strings as following example.
```
            [ 31514] |   gnutls_certificate_set_x509_trust_file() {
   7.234 ms [ 31514] |     /* linux:schedule */
            [ 31514] |     fopen("/etc/ssl/certs/ca-certificates.crt", "rb") {
  10.335 ms [ 31514] |       /* linux:schedule */
   8.292 ms [ 31514] |       /* linux:schedule */
  18.746 ms [ 31514] |     } = 0x555a57c774d0; /* fopen */
   4.376 us [ 31514] |     fileno(0x555a57c774d0) = 6;
   4.180 us [ 31514] |     __fxstat();
   4.704 us [ 31514] |     ftello();
            [ 31514] |     fread(0x7f1f36f1e010, 1, 0x38fb3, 0x555a57c774d0) {
   6.605 ms [ 31514] |       /* linux:schedule */
   6.969 ms [ 31514] |     } = 0x38fb2; /* fread */
   0.564 us [ 31514] |     __errno_location();
   3.713 us [ 31514] |     ferror(0x555a57c774d0) = 0;
   0.207 us [ 31514] |     __errno_location();
  10.983 us [ 31514] |     fclose(0x555a57c774d0) = 0;
 944.565 us [ 31514] |     memmem("-----BEGIN CERTIFICATE-----
MIIH0zCCBbugAwIBAgIIXsO3pkN/pOAwDQYJKoZIhvcNAQEFBQAwQjESMBAGA1UE
Aw...", 0x38fb2, "-----BEGIN CERTIFICATE", 22) = "-----BEGIN CERTIFICATE-----
MIIH0zCCBbugAwIBAgIIXsO3pkN/pOAwDQYJKoZIhvcNAQEFBQAwQjESMBAGA1UE
Aw...";
            [ 31514] |     asn1_create_element() {
   1.420 us [ 31514] |       strchr("PKIX1.Certificate", '.') = ".Certificate";
   0.400 us [ 31514] |       __memcpy_chk();
```
But if `memmem` is used as the original form as `void*` then it's better not to include it in `misc/prototypes.h` because it will just show only the addresses which doesn't look informative to users.

Reply:
I was going to show some strings as following example.
```
            [ 31514] |   gnutls_certificate_set_x509_trust_file() {
   7.234 ms [ 31514] |     /* linux:schedule */
            [ 31514] |     fopen("/etc/ssl/certs/ca-certificates.crt", "rb") {
  10.335 ms [ 31514] |       /* linux:schedule */
   8.292 ms [ 31514] |       /* linux:schedule */
  18.746 ms [ 31514] |     } = 0x555a57c774d0; /* fopen */
   4.376 us [ 31514] |     fileno(0x555a57c774d0) = 6;
   4.180 us [ 31514] |     __fxstat();
   4.704 us [ 31514] |     ftello();
            [ 31514] |     fread(0x7f1f36f1e010, 1, 0x38fb3, 0x555a57c774d0) {
   6.605 ms [ 31514] |       /* linux:schedule */
   6.969 ms [ 31514] |     } = 0x38fb2; /* fread */
   0.564 us [ 31514] |     __errno_location();
   3.713 us [ 31514] |     ferror(0x555a57c774d0) = 0;
   0.207 us [ 31514] |     __errno_location();
  10.983 us [ 31514] |     fclose(0x555a57c774d0) = 0;
 944.565 us [ 31514] |     memmem("-----BEGIN CERTIFICATE-----
MII0zccBbugawIBAgIIXO3pkN/poAwDQYJKoZIhcNAQEFBQAwQjESBG2ue
Aw...", 0x38fb2, "-----BEGIN CERTIFICATE", 22) = "-----BEGIN CERTIFICATE-----
H0zccBbAwIBAgIXsO3pkN/pOAwDQYJoZIhvcNAQEFBQAQjEsMBAGa2ue
Aw...";
            [ 31514] |     asn1_create_element() {
   1.420 us [ 31514] |       strchr("PKIX1.Certificate", '.') = ".Certificate";
   0.400 us [ 31514] |       __memcpy_chk();
```
But if `memmem` is used as the original form as `void*` then it's better not to include it in `misc/prototypes.h` because it will just show only the addresses which doesn't look informative to users.